modules/backend/layouts/_mainmenu.htm

```diff
@@ -24,7 +24,7 @@
                         <a href="<?= $item->url ?>">
                             <span class="nav-icon">
                                 <?php if ($item->iconSvg): ?>
-                                    <img class="svg-icon" src="<?= Url::asset($item->iconSvg) ?>">
+                                    <img role="img" loading="lazy" class="svg-icon" src="<?= Url::asset($item->iconSvg) ?>" alt="<?= e(trans($item->label)) ?>" width="30" height="30">
```

Comment:
why is role="img" required? Also I'm not sure if the alt text should be just the label repeated as screen readers will then be repeating the label of the menu item twice. I'm not sure we should have an alt text for these icons TBH as menu items are required to have labels set AFAIK. If we do decide to have alt text, then it should be in a string like "$label menu item icon" and / or allowed to be overridden by the devs.

Reply:
I removed the role="img" my bad. As for the rest, I'm going to run some tests with a screen reader and see what is going on and will comment back.

Reply:
@LukeTowers see screenshot as proof:

![image](https://user-images.githubusercontent.com/17784082/64266941-6bc41000-cf2d-11e9-95f2-7da76fd54961.png)

So adding alt-tags are needed for a screen reader and they are not displayed twice!

> If we do decide to have alt text, then it should be in a string like "$label menu item icon" and / or allowed to be overridden by the devs.

I think you don't need to do that.

Right now the alt-tag is displaying:

```html
<img loading="lazy" class="svg-icon" src="http://localhost/modules/backend/assets/images/dashboard-icon.svg" alt="Dashboard" width="30" height="30">
```

I plan to create another PR for the top navigation with regards to adding keyboard navigation and screen reader support. So I will be adding extra code that will turn that section into a special container, basically the screen reader will be reading it like this:

"Main menu, contains eight items,.. item one of eight dahsboard,.. etc."

So I think it's best to have the alt-tag changed to this:

```html
alt="<?= e($item->label) ?>"
```

Let me know if you are happy with all that and your thoughts?

Reply:
![image](https://user-images.githubusercontent.com/17784082/64268401-e857ee00-cf2f-11e9-90fb-c1aa107193ab.png)

Removed the alt-tag.

Reply:
@LukeTowers 

Ok done some research and here is the following corrections and results:

### Menu item links

![image](https://user-images.githubusercontent.com/17784082/64268994-07a34b00-cf31-11e9-9144-d9fe841d761d.png)

Solution: Removed the alt-tag, the result is:

```html
Name: "Dashboard"

    aria-labelledby: Not specified
    aria-label: Not specified
    Contents: "Dashboard" <---
    title: Not specified
```

### Preview the website link

![image](https://user-images.githubusercontent.com/17784082/64269309-931cdc00-cf31-11e9-9098-50f8428162c1.png)

Solution: There was no alt-tag, but the result was blank and I've added an `aria-label` attribute that now gives the result for the screen reader, the result is:

```html
Name: "Preview the website"

    aria-labelledby: Not specified
    aria-label: "Preview the website" <---
    Contents: Not specified
    title: Not specified
```

### Avatar link

![image](https://user-images.githubusercontent.com/17784082/64269448-cc554c00-cf31-11e9-8ace-6917ff1e450b.png)

Solution: I've removed the `alt-tag` and I've added an `aria-label` attribute that now gives the result for the screen reader, the result is:


```html
Name: "Signed in as Admin Person" 

    aria-labelledby: Not specified
    aria-label: "Signed in as Admin Person" <---
    Contents: Not specified
    title: Not specified
```

Hope that is fine with you, now passes my screen reader tests.

Reply:
@LukeTowers Have marked this as un-resolved. Please can you give me a suggestion for a better description of this sub-menu for the avatar, see previous comment (sorry I was writing it while I was grabbing each test - maybe you saw an old cached version in github).